src/image/ImageManagerProtocol.cc

```diff
@@ -416,33 +435,107 @@ void ImageManager::_rm(unique_ptr<image_msg_t> msg)
 {
     NebulaLog::dddebug("ImM", "_rm: " + msg->payload());
 
+    int img_id = msg->oid();
     int ds_id = -1;
+    int uid;
+    int gid;
     int rc;
 
     string tmp_error;
     string source;
 
+    long long size;
+    long long snap_size;
+
     ostringstream oss;
 
-    if ( auto image = ipool->get(msg->oid()) )
+    int backup_vm_id = -1;
+
+    bool success = msg->status() == "SUCCESS";
+
+    if ( auto image = ipool->get(img_id) )
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We observed that you are using unintended operations or assignments that could lead to logical errors, vulnerabilities, or undefined behavior. To prevent these issues, ensure that you follow best practices, such as using explicit comparisons and clear operator usage, and using parentheses to clarify intent. Learn more - https://cvefeed.io/cwe/detail/cwe-480-use-of-incorrect-operator